language/move-binary-format/src/deserializer.rs

```diff
@@ -1471,17 +1480,19 @@ fn load_code(cursor: &mut VersionedCursor, code: &mut Vec<Bytecode>) -> BinaryLo
                 Bytecode::MoveToGeneric(load_struct_def_inst_index(cursor)?)
             }
             Opcodes::FREEZE_REF => Bytecode::FreezeRef,
-            Opcodes::VEC_PACK => {
-                Bytecode::VecPack(load_signature_index(cursor)?, read_u64_internal(cursor)?)
-            }
+            Opcodes::VEC_PACK => Bytecode::VecPack(
+                load_signature_index(cursor)?,
+                read_u64_internal_bounded(cursor, VEC_PACK_UNPACK_MAX)?,
+            ),
             Opcodes::VEC_LEN => Bytecode::VecLen(load_signature_index(cursor)?),
             Opcodes::VEC_IMM_BORROW => Bytecode::VecImmBorrow(load_signature_index(cursor)?),
             Opcodes::VEC_MUT_BORROW => Bytecode::VecMutBorrow(load_signature_index(cursor)?),
             Opcodes::VEC_PUSH_BACK => Bytecode::VecPushBack(load_signature_index(cursor)?),
             Opcodes::VEC_POP_BACK => Bytecode::VecPopBack(load_signature_index(cursor)?),
-            Opcodes::VEC_UNPACK => {
-                Bytecode::VecUnpack(load_signature_index(cursor)?, read_u64_internal(cursor)?)
-            }
+            Opcodes::VEC_UNPACK => Bytecode::VecUnpack(
+                load_signature_index(cursor)?,
+                read_u64_internal_bounded(cursor, VEC_PACK_UNPACK_MAX)?,
```

Comment:
Does this mean that the instruction `VecPack[VEC_PACK_UNPACK_MAX + 1]` could be included in published code + serialized, but would fail to deserialize (assuming you could get the published code to pass the verifier)?